site/guides/integration-testing.adoc

```diff
@@ -0,0 +1,1032 @@
+the template uses a conceptually simple mechanism that checks out the application's source code from Github,
+recognizes the application as a Java Maven application, uses Maven to create the project's artifact (the Spring Boot uber-jar) inside a dedicated (so called builder) Pod,
+creates a Docker image inside OpenShift's registry and finally uses it to deploy the application.
+
+For this approach to work, Maven needs to be configured to produce a Spring Boot uber-jar (in the same way as is needed to run the application locally).
+The relevant Maven configuration is the typical Spring Boot uber-jar configuration like so:
+
+[source,xml]
+----
+      <plugin>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+        <configuration/>
+        <executions>
+          <execution>
+            <goals>
+              <goal>repackage</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+----
+
+Just as a reminder we note here that this configured results in the production of two jar files when the Maven `package` phase is executed. OpenShift is able to correctly use the uber-jar
+since the `spring-boot-maven-plugin` takes care to change the suffix of the non uber-jar.
+
+==== Maven Dependencies Configuration
+
+The first order of business is to add Arquillian to the dependencies of the project. The best way to do that is to import the Arquillian BOM to Maven's
+`dependencyManagement` section like so:
+
+[source,xml]
+----
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.jboss.arquillian</groupId>
+        <artifactId>arquillian-bom</artifactId>
+        <version>1.4.0.Final</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+----
+
+Adding the appropriate dependencies then becomes a matter of adding the following configuration:
+
+[source,xml]
+----
+  <dependencies>
+    <dependency>
+      <groupId>org.jboss.arquillian.junit</groupId>
+      <artifactId>arquillian-junit-standalone</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.arquillian.cube</groupId>
+      <artifactId>arquillian-cube-openshift</artifactId>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>io.undertow</groupId>
+          <artifactId>undertow-core</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency
+   <dependencies>
+----
+
+The dependencies above provide add the minimum number of arquillian dependencies needed to interact with the application when it's deployed on OpenShift.
+
+==== Maven failsafe plugin configuration
+
+Making sure that the integration tests only run in a proper environment is very important. We don't want developers inadvertently launching the integration tests
```

Comment:
While separate profile certainly makes sense. A nicer way to make sure that openshift is available is to use a conditional runner:
```
@RequiresOpenshift
@RunWith(ArquillianConditionalRunner.class)
public class MyIT {
....
```

Reply:
Wasn't aware of  this! Thanks, I'll add a note about it